driver/src/orderbook/shadow_orderbook.rs

```diff
@@ -71,6 +71,9 @@ fn background_shadow_reader(
     reader: &dyn StableXOrderBookReading,
     channel: Receiver<(u32, Orderbook)>,
 ) {
+    if let Err(err) = reader.initialize() {
```

Comment:
If initialize had an "optional" semantics it would be fine to not call it here.

Reply:
It's not needed but I we might as well do it.

Reply:
An argument for not doing it would be that this would happen even before a signal to the channel was sent and thus likely happen at the same time (and thus consume node power) as our more critical primary orderbook fetching.

Reply:
For our real configuration this line doesn't do anything anyway because the orderbook we use as the shadow does not do any initialization.
Conceptually it makes more sense to initialize here because we are in another thread after all and we will need this orderbook.

Reply:
I'm fine with this decision, but want to point out that the argument 

> For our real configuration this line doesn't do anything anyway

is very brittle and can easily change in the future without someone remembering to update this place (the shadow orderbook knows nothing about the concrete implementation of Reading).

Reply:
That argument works the other way too. If we had an expensive to initialize orderbook here then it would be a problem to not initialize it early because then the orderbook comparison would start too late.
What I meant by "conceptually" is that if I read the orderbookreading trait and see this initialize method then I would definitely call it in this place in the shadow orderbook.

Reply:
> happen at the same time (and thus consume node power) as our more critical primary orderbook fetching.

Is this still a problem if our node is sitting behind a node balancer? I would be interested in seeing a quick (and un-scientific) benchmark of running an event orderbook shadowed by an event orderbook and compare start up times (just for curiosity's sake).

Also, does excessive querying affect the ability of a node to verify a solution and submit a transaction? If so I'd be more inclined to have a slower initialization then have potential problems querying the node or submitting a transaction when it actually matters.

> the orderbook comparison would start too late.

This shouldn't be a problem since we ignore new comparisons if the previous ones are taking too long. Arguably warming up as fast as possible (and not missing batches) has more value than missing a shadowed orderbook verification (which is just for piece of mind and has no impact to users)

> I read the orderbookreading trait and see this initialize method then I would definitely call it in this place in the shadow orderbook.

This also makes sense to me. To me calling `initialize` is also the "intuitive" way to implement this given the trait description.

Reply:
> I read the orderbookreading trait and see this initialize method then I would definitely call it in this place in the shadow orderbook.

> This also makes sense to me. To me calling initialize is also the "intuitive" way to implement this given the trait description.

To me the intuitive (semi-lazy) way would be to initialize the shadow when someone is calling initialize on the shadowing orderbook itself (if no-one wants to initialize the shadowing orderbook, why should it initialize its sub-orderbooks). 
I see that this might be a bit annoying to implement given the owner structure here.

Reply:
> initialize on the shadowing orderbook

That’s a very good point, I hadn't thought about it.  Sorry if I had misunderstood our previous point.

Maybe the “hidden” expensive background initialization is not the best idea.

> I see that this might be a bit annoying to implement given the owner structure here.

I think it would be fairly easy to do over the channel. It would send a `Request` enum that is either a `Initialize` or a `Verify`.

Reply:
To keep this PR simple and let it accomplish the original goal which was just to initialize the event based orderbook I am going to remove any initialization from the shadow orderbook. We can rethink this later.